### PR TITLE
Add support for token based login

### PR DESCRIPTION
**Pull request by lordievader**
Token based logins allows bots to reuse an earlier login. Instead of
logging in and creating a new session at each start with the risk of
never logging out due to a crash or something.

### CODE REVIEW

File: matrix_bot_api/matrix_bot_api.py [hunk 1]
```diff
@@ -10,20 +10,24 @@ class MatrixBotAPI:
...
-            print(e)
-            if e.code == 403:
-                print("Bad username/password")
-        except Exception as e:
-            print("Invalid server URL")
-            traceback.print_exc()
+        self.client = MatrixClient(server, user_id=username, token=token)
+        if token is None:
+            try:
+                print('logging in with password')
```
Comment (shawnanastasio) [inline matrix_bot_api/matrix_bot_api.py:22]: Did you mean to leave this in, or was it just for local debugging? We probably shouldn't have `print()`s outside of error paths (and even there it's questionable, but existing code already does it...)

Comment (lordievader) [reply matrix_bot_api/matrix_bot_api.py:22]: Ah, no. That should have been removed. Was added for testing purposes.

Comment (lordievader) [reply matrix_bot_api/matrix_bot_api.py]: Statement removed in 68212e50c135970d242378d3c4279a9af2af9aeb.

**Comment by shawnanastasio** [comment]
Merged, thanks!

**Pull request merged**